### ISSUE TITLE
Feature: Bitwise operation derives

### ISSUE THREAD

**Issue by Davoodeh** [JelteF/derive_more#197]
Hello and thanks for the great work to every contributor.

I was thinking, however rare, but maybe bitwise derives are missing for custom flag types (unnamed structs of bools or even vec/array of bools or something like that).
Nothing fancy, but maybe for starters we can have some examples like below:

Proposed derives:
- `BitwiseAnd`
- `BitwiseOr`
- `BitwiseXor`
- `BitwiseBool` = `BitwiseAnd + BitwiseOr + BitwiseXor`

I'll be happy to contribute it myself however I am actually not certain about the quality of my code in Rust yet and also the package seems a bit over complicate and without a newbie friendly contribution guide.

### `BitwiseAnd`
```rust
#[derive(BitwiseAnd)]
struct X(bool, bool, bool);

// The macro generates the code below:
impl std::ops::BitAnd for X {
    type Output = X;

    fn bitor(self, rhs: Self) -> Self::Output {
        Self(
            self.0 | rhs.0,
            self.1 | rhs.1,
            self.2 | rhs.2,
        )
    }
}

impl std::ops::BitAndAssign for X {
    fn bitand_assign(&mut self, rhs: Self) {
        *self = Self(
            self.0 | rhs.0,
            self.1 | rhs.1,
            self.2 | rhs.2,
        )
    }
}
```

**Comment by tyranron** [comment]
@Davoodeh I'm not against having such derives in `derive_more`, but somewhat opposed to `Bitwise*` naming. As for me, it would be nice to preserve granularity and naming from `std`, in the manner we do for other traits.

@JelteF @ilslv what would you say?

**Comment by ilslv** [comment]
@tyranron agreed, I would like for all `derive` macros having exact `std` trait name match.

**Comment by Davoodeh** [comment]
Thank you for responding.

I have no comment on the naming. Just thought I introduce the idea. The rest of the hard work is for the contributors :D

I also implemented a full version here which has support for boolean fields or an array field or a vector field:
https://github.com/Davoodeh/bitwise/blob/master/src/lib.rs

It's messy but might be helpful.

**Comment by Davoodeh** [comment]
Okay, a little while has passed, I have a bit of time on my hand now and I think I can tackle this on my own (and of course with your guidelines).

I'll bump it soon with a PR (hopefully).
